content/scripts/build.js

```diff
@@ -429,72 +437,99 @@ class Builder {
+    let t0 = new Date();
+    this.sources.entries().forEach(source => {
+      this.getLocaleRootFolders(source, { allLocales: true }).forEach(
+        filepath => {
+          walker(filepath, (folder, files) => {
+            if (source.isStumptown) {
+              files
+                .filter(n => n.endsWith(".json"))
+                .forEach(filename => {
+                  const filepath = path.join(folder, filename);
+                  this.processStumptownFileTitle(source, filepath);
+                });
+              return;
+            }
+            if (files.includes("index.html") && files.includes("index.yaml")) {
+              this.processFolderTitle(source, folder);
+            }
+          });
+        }
       );
-    }
-  }
+    });
 
-  ensureAllChildren() {
-    this.ensureAllTitles();
-    Object.entries(this.allTitles).forEach(([uri, data]) => {
-      if (data.parent) {
-        const parentMdnURL = buildMDNUrl(data.parent.locale, data.parent.slug);
-        if (this.allTitles[parentMdnURL]) {
-          if (!(parentMdnURL in this.allChildren)) {
-            this.allChildren[parentMdnURL] = [];
+    // Only after *all* titles have been processed can we iterate over the
+    // mapping and figure out all translations.
+    // What this does is that it sets `.translations=[{slug, locale}, ...]`
+    // on every title that is the "translation_of". Practically, that means
+    // that every 'en-US' document that has been translated, will have a list
+    // of other locales and slugs.
+    let countBrokenTranslationOfDocuments = 0;
+    Object.values(this.allTitles)
+      .filter(data => data.translation_of)
+      .forEach(data => {
+        const parentURL = buildMDNUrl("en-US", data.translation_of);
+        const parentData = this.allTitles[parentURL];
+
+        // TODO: Our dumper is not perfect yet. We still get bad
+        // 'translation_of' references.
+        // I.e. a localized document's 'index.yaml' says its
+        // 'translation_of' is 'Web/Foo/Bar' but there is actually no en-US
+        // document by that slug!
+        // We're working on it in the dumper and this problem is known also
+        // in the 'ensureAllTitles()' method which at least debug logs the
+        // bad ones and warn logs about a total count of bad documents.
+        // Once the dumper has matured, we'll remove this defensive style and
+        // throw an error here. That'll be a form of validation-by-building
+        // which can really help our CI trap content edit PRs that sets
+        // or gets these references wrong.
+
+        if (parentData) {
+          if (!("translations" in parentData)) {
+            parentData.translations = [];
           }
-          this.allChildren[parentMdnURL].push({
+          parentData.translations.push({
             locale: data.locale,
             slug: data.slug
           });
+        } else {
+          countBrokenTranslationOfDocuments++;
```

Comment:
This has to eventually disappear. We'll hopefully soon fix our dumper so that this number becomes 0. When that's the case I think we should do something like this (a couple of lines higher up here)
```diff
        const parentData = this.allTitles[parentURL];
+      if (!parentData) {
+        throw new Error(`Build error in ${data.file} (${data.locale}/${data.slug}). Invalid 'translation_of' reference. Bla bla`);
+     }
```

That'll make sure nobody accidentally messes with any `index.yaml` 's en-US `slug` or any 'translation_of' without using the proper yari tools for doing renamings. And this'll break the CI build which'll quickly inform the PR that it's potentially a corrupted change.